viam_cartographer.go

```diff
@@ -214,58 +207,47 @@ func New(
 	}
 
 	// Get the movement sensor if one is configured and check if it supports an IMU and/or odometer.
-	var imuSupported, odometerSupported bool
+	var timedMovementSensor s.TimedMovementSensor
 	if movementSensorName == "" {
 		logger.Info("no movement sensor configured, proceeding without IMU and without odometer")
 	} else {
-		imuSupported, odometerSupported, err = checkIfIMUAndOdometerSupported(ctx, deps, movementSensorName)
-	}
-
-	// Get the IMU if it is supported
-	var timedIMU s.TimedIMUSensor
-	if imuSupported {
-		timedIMU, err = s.NewIMU(ctx, deps, movementSensorName, optionalConfigParams.MovementSensorDataFrequencyHz, logger)
-		if err != nil {
+		if err = checkIfIMUAndOdometerSupported(ctx, deps, movementSensorName); err != nil {
```

Comment:
[nit] I think this check is redundant with the check in NewMovementSensor() and can be removed.

Reply:
Agreed